vpr/src/route/rr_graph_uxsdcxx_serializer.h

```diff
@@ -1567,148 +1567,50 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
-            } else if (rr_graph.node_type(node.id()) == IPIN || rr_graph.node_type(node.id()) == OPIN) {
-                for (int ix = node.xlow(); ix <= node.xhigh(); ix++) {
-                    for (int iy = node.ylow(); iy <= node.yhigh(); iy++) {
-                        for (const e_side& side : SIDES) {
-                            if (!node.is_node_on_specific_side(side)) {
-                                continue;
-                            }
-                            if (node.ptc_num() >= (int)indices[OPIN][ix][iy][side].size()) {
-                                indices[OPIN][ix][iy][side].resize(node.ptc_num() + 1, OPEN);
-                            }
-                            if (node.ptc_num() >= (int)indices[IPIN][ix][iy][side].size()) {
-                                indices[IPIN][ix][iy][side].resize(node.ptc_num() + 1, OPEN);
-                            }
-                            indices[rr_graph.node_type(node.id())][ix][iy][side][node.ptc_num()] = inode;
-                        }
-                    }
-                }
-            } else if (rr_graph.node_type(node.id()) == CHANX) {
-                for (int ix = node.xlow(); ix <= node.xhigh(); ix++) {
-                    for (int iy = node.ylow(); iy <= node.yhigh(); iy++) {
-                        coordinates_max_ptc[iy][ix].chanx_max_ptc = std::max(coordinates_max_ptc[iy][ix].chanx_max_ptc, node.ptc_num());
-                    }
-                }
-            } else if (rr_graph.node_type(node.id()) == CHANY) {
-                for (int ix = node.xlow(); ix <= node.xhigh(); ix++) {
-                    for (int iy = node.ylow(); iy <= node.yhigh(); iy++) {
-                        coordinates_max_ptc[ix][iy].chany_max_ptc = std::max(coordinates_max_ptc[ix][iy].chany_max_ptc, node.ptc_num());
-                    }
-                }
-            }
-        }
-
-        /* Alloc the lookup table */
-        for (t_rr_type rr_type : RR_TYPES) {
-            if (rr_type == CHANX) {
-                for (size_t y = 0; y < grid_.height(); ++y) {
-                    for (size_t x = 0; x < grid_.width(); ++x) {
-                        indices[CHANX][y][x][0].resize(coordinates_max_ptc[y][x].chanx_max_ptc + 1, OPEN);
-                    }
-                }
-            } else if (rr_type == CHANY) {
-                for (size_t x = 0; x < grid_.width(); ++x) {
-                    for (size_t y = 0; y < grid_.height(); ++y) {
-                        indices[CHANY][x][y][0].resize(coordinates_max_ptc[x][y].chany_max_ptc + 1, OPEN);
-                    }
-                }
-            }
-        }
-
-        int count;
-        /* CHANX and CHANY need to reevaluated with its ptc num as the correct index*/
-        for (size_t inode = 0; inode < rr_nodes_->size(); inode++) {
-            auto node = (*rr_nodes_)[inode];
-            if (rr_graph.node_type(node.id()) == CHANX) {
+            for (int ix = node.xlow(); ix <= node.xhigh(); ix++) {
```

Comment:
Could all this be made a utility function in RRGraphBuilder (or a global utility function) and called in multiple places?  This code would handle the 3 call sites with 2D loop nests in the clock rr-graph builder too.  It seems like making this a general utility would be a good idea.  Would be a function like: add_node_to_all_locs () or some such.

Reply:
Agree. The node addition is very generic and applicable to all kinds of nodes. I think it should be an API of the RRGraphBuilder because it should be the only way to add an existing ``rr_node`` to RRSpatialLookup data structure.